sopel/bot.py

```diff
@@ -592,6 +592,71 @@ def register_urls(self, urls: Iterable) -> None:
                 except plugins.exceptions.PluginError as err:
                     LOGGER.error("Cannot register URL callback: %s", err)
 
+    def rate_limit_info(self, rule: plugin_rules.Rule, trigger: Trigger) -> Tuple[bool, Optional[str]]:
+        if trigger.admin or rule.is_unblockable():
+            return False, None
+
+        is_channel = trigger.sender and not trigger.sender.is_nick()
+        channel = trigger.sender if is_channel else None
```

Comment:
Given the usage of `channel` in the `format()` down the line, I'd replace it by something that can be read, like `private-message`:

```suggestion
        channel = trigger.sender if is_channel else 'private-message'
```

The result would look like this:

```python
>>> 'You must wait before using this command in {channel} again.'.format(channel='private-message')
'You must wait before using this command in private-message again.'
```

That should be OK in most cases. Note that there is one and only one occasion to have this "private-message" is when the user is directly calling the bot in private.